Bump ZeroMQ and fix some Buffer deprecations

mauricioszabo opened the pull request:
I found by trial and error that ZeroMQ 6 works with the latest Electron version, but unfortunately not the version that's present on this branch.

This PR bumps the version, and fixes deprecation warnings about Buffer().

mauricioszabo committed:
Fixed Buffer deprecations

mauricioszabo committed:
Bump ZeroMQ

mauricioszabo committed:
Adding a lockfile

n-riesco commented:
Hey! Thank you for doing this. PR looks good and it only needs addressing the points below:

- [ ] remove the lockfile (I don't distibute lockfiles for libraries. Library users should carry the burden of maintaining their own. On my end, I'll make sure `jmp` abides by semver and update the version numbers and `package.json` whenever any our dependencies break semver.
- [ ] please, read https://github.com/n-riesco/jmp/blob/master/CONTRIBUTING.md#code-contributions and add yourself to AUTHORS to acknowledge the license terms.
- [ ] I know `zeromq.js` hasn't bumped the engine requirements, but on my side, I want to test this PR locally, because often `zeromq.js`'s dependencies do so silently.

mauricioszabo committed:
Remove lockfile

mauricioszabo committed:
Added Maurício to AUTHORS file.

mauricioszabo commented:
Hi, @n-riesco, sorry for the delay, kinda forgot about this PR that I opened.

So, I removed the lockfile and added myself to the authors. Now, I don't understand what the last point is, can you clarify for me please? Which engine do you mention?

And also, thanks for taking a look at this PR :).

n-riesco commented:
@mauricioszabo By node engine, I mean node version. See https://docs.npmjs.com/cli/v10/configuring-npm/package-json#engines .
